### ISSUE TITLE
cylc review: Server 404 error when trying Display Options for suites contain special characters

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-flow#3119]
Hi, I am testing some suites with special characters in Cylc Review, and found that when using:

```
cylc register  "''.__abc__.__def__[2]()[40]('/pao/com/manteira')"  .
```

where the suite name is `''.__abc__.__def__[2]()[40]('/pao/com/manteira')`, it is registered and runs fine in Cylc, creating logs, running the tasks all right.

But in Cylc Review I get errors when doing the following:

1. Open Cylc Review and type user name to list suites
2. Choose "cycles list" for the `''.__abc__.__def__[2]()[40]('/pao/com/manteira')` suite
3. Expand "Display Options" to customize how items are displayed in the page
4. Do not type or touch anything, just press "Update"

You will get a 404, because the server is rewriting the URL query string and parameters, which does not look like the URL for a suite like "cylc1".

![image](https://user-images.githubusercontent.com/304786/56548213-a1e2a200-65d3-11e9-8fdd-d89e3cbf8089.png)

**Comment by sadielbartholomew** [comment]
> 2. Choose "cycles list" for the ''.__abc__.__def__[2]()[40]('/pao/com/manteira') suite

Just to note that this bug is also there on the "task jobs list" as well as the "cycles list" page as outlined, for suites & use cases identical to those described for the latter case.

**Comment by hjoliver** [comment]
Closed by #3125 (did not auto-close?)

**Comment by kinow** [comment]
I think it might not auto close if the pull request was merged against a branch that is not the default branch `master`.
